connectors/sources/mysql.py

```diff
@@ -21,12 +27,92 @@
     "TABLE_LAST_UPDATE_TIME": "SELECT UPDATE_TIME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = '{database}' AND TABLE_NAME = '{table}'",
 }
 DEFAULT_FETCH_SIZE = 50
-DEFAULT_RETRY_COUNT = 3
-DEFAULT_WAIT_MULTIPLIER = 2
+RETRIES = 3
+RETRY_INTERVAL = 2
 DEFAULT_SSL_DISABLED = True
 DEFAULT_SSL_CA = None
 
 
+def format_list(databases):
+    return ", ".join(databases)
+
+
+class MySQLAdvancedRulesValidator(AdvancedRulesValidator):
+    def __init__(self, source):
+        self.source = source
+
+    async def validate(self, advanced_rules):
+        databases = set(self.source.configured_databases())
+        databases_to_filter = set(advanced_rules.keys())
+
+        missing_databases = list(databases_to_filter - databases)
+
+        if len(missing_databases) > 0:
+            return SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=f"Non-configured databases: {format_list(missing_databases)}. Configured databases: {'None' if not len(databases) else format_list(databases)}.",
+            )
+
+        return await self._remote_validation(advanced_rules, databases_to_filter)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def _remote_validation(self, advanced_rules, databases_to_filter):
+        await self.source.ping()
+
+        inaccessible_databases = set(
+            await self.source.validate_databases(databases=databases_to_filter)
+        )
+        inaccessible_databases_to_filter = inaccessible_databases.intersection(
+            databases_to_filter
+        )
+
+        if len(inaccessible_databases_to_filter) > 0:
+            return SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=f"Inaccessible databases: {format_list(inaccessible_databases_to_filter)} for user '{self.source.configuration.get('user', 'None')}'.",
```

Comment:
do we really want to display `None` for the user by default?

Reply:
Adjusted it to "No user configured"